spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -276,13 +286,21 @@ else if (messageListener instanceof MessageListener) {
 			throw new IllegalStateException("messageListener must be 'MessageListener' "
 					+ "or 'AcknowledgingMessageListener', not " + messageListener.getClass().getName());
 		}
-		if (getTaskExecutor() == null) {
-			setTaskExecutor(
-					new SimpleAsyncTaskExecutor(getBeanName() == null ? "kafka-" : (getBeanName() + "-kafka-")));
+		if (getConsumerTaskExecutor() == null) {
+			SimpleAsyncTaskExecutor pooledExecutor = new SimpleAsyncTaskExecutor(
```

Comment:
The `SimpleAsyncTaskExecutor` isn't  `pooled`:
```
* <p><b>NOTE: This implementation does not reuse threads!</b> Consider a
* thread-pooling TaskExecutor implementation instead, in particular for
* executing a large number of short-lived tasks.
```
Maybe we really should reconsider in favor of `ThreadPoolTaskExecutor` ?